packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1391,22 +1391,60 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction hash associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param hash - The new hash value to be assigned.
+   */
+  updateTransactionHash(transactionId: string, hash: string) {
```

Comment:
As discussed, rather than exposing these for general usage, shall we instead keep them as specific as possible in order to protect the state?

This could become `updateCustodialTransactionHash` which validates the transaction has a `custodyId` and the correct state such as `submitted`?